### DISCUSSION TITLE
Any endorsed machine learning library?

### DISCUSSION THREAD

**Discussion by c0c0n3** [c0c0n3/kitt4sme.live#143] [Q&A] [answered]
Cross posting the question @pmpoukli [asked in the private forum][pvt-forum] since it could be useful to others:

> (1) Is there any existing library in KITT4SME which can be made available for experiments in order to review it? (In order to check the logic behind it). 
> Also, if there is, (2) is it obligatory to use this library instead of another one?

[pvt-forum]: https://github.com/kittOC/FAQs-and-Technical-issues/discussions/3

**Comment by c0c0n3** [comment, marked as answer]
copy-pasting answer form private forum below.

Hi @pmpoukli!

We're not endorsing any particular machine learning library or requiring open call winners to use a specific one. You should use whatever library makes sense for your use case---e.g. scikit-learn (Python), MLlib (Scala, Java), etc.

What we [initially hoped to do][ai-frameworks] was to include popular machine learning runtimes (e.g. TensorFlow) in the [platform infra services layer][plat.dia] to avoid everyone having to deploy the runtime separately---a common infra means less dev, management, and maintenance overhead, not to mention possible optimisation of compute resources. At the moment we're not sure we'll have the resources to do that, but if most of you think you could share the same runtime infra, then we should have a chat about it to avoid everyone speding dev cycles on the same task!

In all fairness there's a couple of libraries we're developing but I think they're only marginally relevant to your question:

- [FiPy][fipy]. We have some initial support for retrieving time series in a frame-friendly format---e.g. something you can readily use to instantiate a Pandas data frame.
- [Dazzler][dazzler]. Dash-powered visualization where you could easily take advantage of scikit-learn to perform further analysis before presenting the data to the user---e.g. look at this [k-means clustering example][kmeans].

Like I said I don't think this is very relevant to your question, but if you think it is, I'll gladly provide more details.



[ai-frameworks]: https://github.com/c0c0n3/kitt4sme.live/issues/54
[dazzler]: https://github.com/c0c0n3/kitt4sme.dazzler
[fipy]: https://github.com/c0c0n3/kitt4sme.fipy
[kmeans]: https://dash-bootstrap-components.opensource.faculty.ai/examples/iris/
[plat.dia]: https://raw.githubusercontent.com/c0c0n3/kitt4sme.live/main/docs/tech-stack.svg
